### ISSUE TITLE
Cannot read properties of undefined (reading 'version')

### ISSUE THREAD

**Issue by wahguzman** [microsoft/AzureStorageExplorer#6399]
No se pueden recuperar los recursos secundarios.

Detalles:
{
  "name": "TypeError",
  "message": "Cannot read properties of undefined (reading 'version')",
  "stack": "TypeError: Cannot read properties of undefined (reading 'version')\n    at connectionIsV1 (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\common\\CustomConnections\\IConnectionInfo.js:8:24)\n    at C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\common\\CustomConnections\\CustomConnectionStore.js:235:74\n    at Array.filter (<anonymous>)\n    at #sendTelemetry (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\common\\CustomConnections\\CustomConnectionStore.js:235:14)\n    at async CustomConnectionStore.getCustomConnectionStore (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\common\\CustomConnections\\CustomConnectionStore.js:44:9)\n    at async PersistentStorage.CustomConnections.getCustomConnections (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\providers\\CustomConnectionsProvider.js:10:39)\n    at async Provider.executeOperation (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\providers\\Provider.js:28:16)\n    at async ElectronRendererHostProxy._handleExecuteRequest (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\ElectronRendererHostProxy.js:213:35)\n    at async ElectronRendererHostProxy._handleMessage (C:\\Program Files (x86)\\Microsoft Azure Storage Explorer\\resources\\app\\out\\app\\renderer\\ElectronRendererHostProxy.js:156:24)"
}

**Comment by geffchang** [comment]
I got this error on 1.27.1. I couldn't see my connecions.

I downgraded to 1.27.0 and all my connections can now be seen. But, all connections are now broken. Very annoying.

**Comment by KristapsD** [comment]
> I got this error on v1.27.1, when I tried to open my connections.
> 
> I downgraded to v1.27.0 and all my connections can now be seen. But, all connections are now broken. I have to re-add all connections.

Same issue, downgrading to 1.27.0 fixes it for me aswell.

**Comment by craxal** [comment]
@wahguzman @geffchang @KristapsD Which version were you using before upgrading to 1.27.1?

Please try the following steps in 1.27.1 and share your findings here to help us look into the problem further:
1. Reproduce the error. 
2. Press F12 to open the dev window.
3. Select the Application tab, then select **Local Storage > file://**.
4. Examine the values for various `StorageExplorer_CustomConnections_*` keys. They should be arrays with values that look something like this:
   ```json
    { "id": "connection-1", "displayName": "Attachment 1" }
   ```
   If the values don't look like this, share what the values look like here.
5. Select the Sources tab, press Ctrl+P, then enter CustomConnectionStore.js.
6. Drop a breakpoint on line 59, which should look like this:
   ```json
   return connectionInfoForAllLocalStgKeys;
   ```
7. Reproduce the error again. The breakpoint should be hit. Examine the value for `connectionInfoForAllLocal`. If you see any undefined values, share that here.

**Comment by craxal** [comment]
We may have traced the issue down to missing data from the credential manager. Please follow these steps to help us verify:

1. Press F12 to open the dev window.
2. Select the Application tab, then select **Local Storage > file://**.
3. Find all the connection IDs stored in various `StorageExplorer_CustomConnections_*` keys. They should be in arrays that look something like this:
   ```json
   [{ "id": "connection-1", "displayName": "Attachment 1" }]
   ```
4. Open your system's credential manager (Credential Manager for Windows, Keychain for mac, various for Linux).
5. Locate the secret corresponding to each connection ID you found in step 3.
   * On Windows, if a table connection has ID `connection-1`, the credential manager should have an entry with key `StorageExplorer_CustomConnections_Tables_v1_connection-1`.
   * On macOS, if a table connection as ID `connection-1`, the credential manager should have a an entry with a service of `StorageExplorer_CustomConnections_Tables_v1` and an account of `connection-1`.

If your credential manager is missing an entry for one or more of your connections, please let us know here.

As a workaround, you can remove any connections that have lost their corresponding data in the credential manager by editing the value or values you found in **Local Storage > file://**.

**Comment by gbpcor** [comment]
@craxal Manually cleaning the list of registered connections fixes the issue.

For me, the list had items without credentials because a network / domain / Windows (?) issue forced me to completely clear out Credential Manager and I didn't clear out those registrations from Storage Explorer. (I used to have ~40 registrations and I kept the broken ones to remind myself to re-register them but I only had time to do the most important ones since this happened.)

Thank you for the workaround.

PS: I know this has nothing to do with you but please pass this along to the people it concerns: the Credential Manager user interface in Windows 10 is one of the most atrocious, badly designed software that I encountered in my career. I know there's always lower but it seems to me that the people involved in making this UI _really_, _truly_ tried to reach new lows with how unusable it is.

**Comment by craxal** [comment]
We have merged some changes that should at the very least allow you to continue working with attachments whose data is still intact; corrupted or missing data should not interfere with working attachments with this change.

**Comment by craxal** [comment]
We've verified the code changes, which will be available in our 1.27.2 release. If you install this version and continue to encounter this issue, please reply with your findings here, and we will look into it further.
